src/rustup-win-installer/build.rs

```diff
@@ -1,13 +1,42 @@
+extern crate gcc;
+
 use std::env;
 
 fn main() {
-    println!("cargo:rustc-link-lib=static=wcautil");
-    println!("cargo:rustc-link-lib=static=dutil");
     println!("cargo:rustc-link-lib=dylib=msi");
     println!("cargo:rustc-link-lib=dylib=user32");
     println!("cargo:rustc-link-lib=dylib=mincore");
 
-    let wix_path = env::var("WIX").unwrap();
-    // x86 target is hard-coded because we only build an x86 installer (works just fine on x64)
-    println!("cargo:rustc-link-search=native={}SDK\\VS2015\\lib\\x86", wix_path);
+    // Part of WIX SDK
+    println!("cargo:rustc-link-lib=static=wcautil");
+    println!("cargo:rustc-link-lib=static=dutil");
+
+    let wix_path = env::var("WIX").expect("WIX must be installed, and 'WIX' environment variable must be set");
+
+    // For the correct WIX library path, we need to know which VS version we are using.
+    // We use the `gcc` crate to get the path to the correct `cl.exe`, and then try to
+    // guess the version from the path components (we don't depend on the actual location
+    // of the VS installation, but only on the internal directory structure).
+    let config = gcc::Config::new();
+    let compiler = config.get_compiler();
+    let path = compiler.path();
+    let vs_version = if path.to_string_lossy().contains("VC\\bin") {
+        println!("cargo:warning=Guessing VS version: VS2015");
+        "VS2015"
+    } else if path.to_string_lossy().contains("VC\\Tools\\MSVC") {
+        println!("cargo:warning=Guessing VS version: VS2017");
+        "VS2017"
+    } else {
+        panic!("Unknown VS version with compiler path {:?}", path);
+    };
```

Comment:
Could you use find_vs_version instead? https://docs.rs/gcc/0.3.51/gcc/windows_registry/fn.find_vs_version.html